### PR TITLE
WIP Tubular Chicken Tubes

### LINKED ISSUE #3252: Minor sea chicken improvements
Chickens were never designed for sea and ultimately these won't change that but they seem like a low mana way to improve the situation:
 - [ ] give chicken projectiles the ability to pierce water.
 - [ ] probably make melee chickens walk underwater (because the juicier eco buildings don't float)
 - [ ] #sonarLives
 - [ ] there are reports that chickens have some sort of slowdown on water, check whether this is true and remove.

### PR DESCRIPTION
**Pull request by ArchShaman**
Hopefully fixes #3252 .

### CODE REVIEW

**Comment by Anarchid** [comment]
This is in serious need of squashing. Also try to not make 24 commits with nearly identical messages next time - at least mention the unit you're changing on the first line.

**Comment by ArchShaman** [comment]
Okay. Throw out and redo?

**Comment by sprunk** [comment]
nah, I can squash on merge

File: units/chicken.lua [hunk 1]
```diff
@@ -44,16 +44,16 @@ unitDef = {
     },
 
   },
-  sightDistance       = 256,
-  sonarDistance       = 200,
+  sightDistance       = 500,
```
Comment (sprunk) [inline units/chicken.lua]: Don't change the sight value, just match the sonar

File: units/chicken.lua [hunk 2]
```diff
@@ -44,16 +44,16 @@ unitDef = {
...
+  sonarDistance       = 500,
   trackOffset         = 0,
   trackStrength       = 8,
   trackStretch        = 1,
   trackType           = [[ChickenTrack]],
   trackWidth          = 18,
   turnRate            = 806,
   upright             = false,
-  waterline           = 16,
+  --waterline           = 16,
```
Comment (sprunk) [inline units/chicken.lua]: Don't leave garbage lines, remove

File: units/chicken_dragon.lua [hunk 3]
```diff
@@ -35,7 +35,7 @@ unitDef = {
   maxSlope               = 36,
   maxVelocity            = 2.1,
   minCloakDistance       = 225,
-  movementClass          = [[AKBOT6]],
+  movementClass          = [[BHOVER3]],
```
Comment (sprunk) [inline units/chicken_dragon.lua]: Use a 5x5 sized class

File: units/chicken_tiamat.lua [hunk 4]
```diff
@@ -34,7 +34,7 @@ unitDef = {
   maxVelocity         = 2.3,
   maxWaterDepth       = 5000,
   minCloakDistance    = 75,
-  movementClass       = [[AKBOT6]],
+  movementClass       = [[AKBOT2]],
```
Comment (sprunk) [inline units/chicken_tiamat.lua]: Don't use a 2x2 class, the unit is 4x4.

File: units/chicken_dragon.lua (hunk 3 above)
Comment (ArchShaman) [reply units/chicken_dragon.lua]: Doesn't exist for hovers, creating one.

**Comment by ArchShaman** [comment]
http://zero-k.info/Forum/Post/191626#191626

Based on the information in this post, I'd like to bring the spires up to par a bit and make them a lot more dangerous (for a late game setting):
* Range 4500 -> 6300
* Projectiles 1 -> 4
* Reload 10 -> 14
* Projectile speed 400 -> 750
    - DPS: 100 -> 285.71

This change would put Spire on par with a big bertha in terms of danger.

Another change I'd like to make (if approved) would be to increase the defensiblity of tubes. Tubes currently have a meager 20 dps (Worse than defender). They feel unimpactful, especially late game. I'd like to make the following changes:


* Explosive spores:
  * Damage: 60x4 -> 100x4
  * Reload time: 12 -> 5
  - DPS: 20 -> 80 (about on par with lotus)

* Add AA weapon to tubes:
  - Damage: 120
  - Reload: 1
  - Range: 1000
  - DPS: 120

The inclusion of a longer ranged AA weapon would help chickens fight air better. Right now, they have nothing that can defend their bases very well against massed air.

**Comment by Histidine91** [comment]
Tube and Spire changes sound fine to me.

Perhaps tube AA should be a flak-type AoE weapon, so large groups of air units take proportionally more damage than small ones?

**Comment by ArchShaman** [comment]
Explosive AA spores? Hm, probably a nice idea. Ideally this will give chickens some response to brawler masses.

**Comment by ArchShaman** [comment]
TO DO:

- [ ] Finalize new chickend unit def
- [ ] Finalize new spire unitdef
- [ ] Testing all unitdefs for weird behavior.

File: scripts/chickenspire.lua [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+local body = piece("body")
+local firepoint = piece("firepoint")
+
+local SIG_AIM = 2 -- no idea what this is for, it was in the original unitscript.
+
+function script.Create()
+  --Normally there's some crap here that randomly sets a direction. Unfortunately I couldn't get this to work in my tests.
+	EmitSfx(body, 1024+2)
+	Move(body, y_axis, -80)
+	Move(body, y_axis, 0,30)
```
Comment (sprunk) [inline scripts/chickenspire.lua]: Different values than the .bos had

File: scripts/chickenspire.lua [hunk 6]
```diff
@@ -0,0 +1,34 @@
+include 'constants.lua'
+
+local body = piece("body")
+local firepoint = piece("firepoint")
+
+local SIG_AIM = 2 -- no idea what this is for, it was in the original unitscript.
+
+function script.Create()
+  --Normally there's some crap here that randomly sets a direction. Unfortunately I couldn't get this to work in my tests.
```
Comment (sprunk) [inline scripts/chickenspire.lua]: `Turn (body, y_axis, math.random() * 2 * math.pi)`

File: scripts/chickenspire.lua [hunk 7]
```diff
@@ -0,0 +1,34 @@
+include 'constants.lua'
+
+local body = piece("body")
+local firepoint = piece("firepoint")
+
+local SIG_AIM = 2 -- no idea what this is for, it was in the original unitscript.
```
Comment (sprunk) [inline scripts/chickenspire.lua]: remove

File: units/chicken_roc.lua [hunk 8]
```diff
@@ -166,7 +166,7 @@ unitDef = {
       tracks                  = true,
       turnRate                = 24000,
       turret                  = true,
-      waterweapon             = true,
+      waterWeapon             = true,
```
Comment (sprunk) [inline units/chicken_roc.lua]: does nothing

File: scripts/chickend.lua [hunk 9]
```diff
@@ -0,0 +1,31 @@
+include 'constants.lua'
+
+local body = piece("body")
+local firepoint = piece("firepoint")
+
+function script.Create()
+	EmitSfx(body, 1024+2)
+	Move(body, y_axis, -80)
```
Comment (sprunk) [inline scripts/chickend.lua]: different value to what was in bos

File: units/chickenspire.lua [hunk 10]
```diff
@@ -136,8 +138,9 @@ unitDef = {
       impulseFactor           = 0.4,
       interceptedByShieldType = 2,
       model                   = [[chickenegggreen_big.s3o]],
-      range                   = 4500,
-      reloadtime              = 10,
+      projectiles	      = 4,
```
Comment (sprunk) [inline units/chickenspire.lua]: dont use tabs for alignment

**Comment by sprunk** [comment]
IMO split the PR into sea and tubes parts (sea looks finalized so could be merged but not when there's the WIP tube stuff)

File: scripts/chickenspire.lua (hunk 6 above)
Comment (ArchShaman) [reply scripts/chickenspire.lua]: Thanks, I'll change it when I get home in two hours.

File: scripts/chickenspire.lua (hunk 5 above)
Comment (ArchShaman) [reply scripts/chickenspire.lua]: Yes sir. They make it so it starts underground.

File: scripts/chickend.lua (hunk 9 above)
Comment (ArchShaman) [reply scripts/chickend.lua]: The -80 makes it so that it starts underground completely.

**Comment by ArchShaman** [comment]
Alright, I'll make changes based on your review and move it from WIP to ready.

The tube update is nearing ready. I just need to find a sound for the new weapon. I have it pretty okay balance wise.
The spire update may take longer.

File: units/chicken_roc.lua (hunk 8 above)
Comment (ArchShaman) [reply units/chicken_roc.lua]: Oh, right because of the lowerkeys at the end.

Comment (ArchShaman) [reply units/chicken_roc.lua]: Do you want it reverted?

**Comment by ArchShaman** [comment]
Any last requests Sprunk before I move it from WIP -> ready ?

File: units/chicken_roc.lua (hunk 8 above)
Comment (sprunk) [reply units/chicken_roc.lua]: Yes (but I can do that myself when merging so don't worry). Not touching stuff needlessly makes tracking history with `git blame` easier to follow and patches easier to review, in this particular case it's not really important but it's generally a good practice to follow.

Comment (ArchShaman) [reply units/chicken_roc.lua]: Ah, okay. I'll keep this in mind next time.

**Comment by ArchShaman** [comment]
Assuming no other complaints, I've move this from WIP to mergeable. I'll do some testing when I get the chance.

**Comment by sprunk** [comment]
I did a rudimentary test and discovered that some of the sonars are not working (eg Cockatrice) and that some chickens still can't path on sea (eg Basilisk).

**Comment by ArchShaman** [comment]
I'll fix that today.

**Comment by ArchShaman** [comment]
Found the issue. Units with sonarDistance need `activateWhenBuilt = true,`

**Comment by Histidine91** [comment]
Does Dragon walk on water? IMO it should walk on sea bottom even, if that means some of its weapons don't work. Or make it dog paddle. 

(It looks silly for something so big to walk on water like giant Jesus lizard, is what I'm saying)

**Comment by ArchShaman** [comment]
I fixed that just now. It now dogpaddles.

**Comment by ArchShaman** [comment]
Bonus fix included: wurms no longer cloak underwater.

**Comment by sprunk** [comment]
I merged sea and spire.

**Comment by ArchShaman** [comment]
Just in case you didn't see it on discord, you'll need to commit a quick change to the tube unitdef to add 'script= [[chickend.lua]]' since you merged the script changes in

**Comment by ArchShaman** [comment]
Name changed since all that's left of the chicken enhancements is the tubes.

**Comment by ArchShaman** [comment]
New enhancements seem to work really well.

**Comment by Anarchid** [comment]
I've made a model for advanced chicken flak. It is here: https://github.com/Anarchid/Zero-K/tree/chickenflak

Screenshot: 
image

**Comment by sprunk** [comment]
Reuse this model for chicken nuke (with an opening animation and all)? I think if used alongside the tube it would bring a lot of attention to how shitty the tube model is.

**Pull request closed without merging**